gamemodes/terrortown/entities/weapons/weapon_zm_carry.lua

```diff
@@ -182,19 +163,21 @@ function SWEP:Reset(keep_velocity)
 		self.Constr:Remove()
 	end
 
-	if IsValid(self.EntHolding) then
+	local ctarget = self:GetCarryTarget()
+	local ctype = self:GetCarryType()
+	if IsValid(ctarget) then
 		-- it is possible for weapons to be already equipped at this point
 		-- changing the owner in such a case would cause problems
-		if not self.EntHolding:IsWeapon() then
+		if ctype ~= CARRY_TYPE_WEAPON then
```

Comment:
Just out for curiosity, what was this changed? Has it any benefit?

Reply:
it reduces a function call every time, since we already ran IsWeapon when we do DetermineCarryType, so it makes sense to me to just continue using that result since they're mutually exclusive -- it's also extremely unlikely but it also guards against IsWeapon being undefined